modules/metadata/CMakeLists.txt

```diff
@@ -17,16 +17,26 @@
 add_ca_library(md_api STATIC
     ${CMAKE_CURRENT_SOURCE_DIR}/source/metadata/utils.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/source/metadata/metadata.cpp)
-target_link_libraries(md_api PUBLIC cargo)
-target_include_directories(md_api
-    PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include
-    PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include/metadata/detail)
-
 add_ca_library(md_handler STATIC
-    ${CMAKE_CURRENT_SOURCE_DIR}/source/metadata/handler/generic_metadata.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/source/metadata/handler/vectorize_info_metadata.cpp)
-target_link_libraries(md_handler PUBLIC cargo md_api)
-target_include_directories(md_handler PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
+  ${CMAKE_CURRENT_SOURCE_DIR}/source/metadata/handler/generic_metadata.cpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/source/metadata/handler/vectorize_info_metadata.cpp)
+if(CA_NATIVE_CPU)
```

Comment:
most of the if/else is the same, maybe refactor to just show the difference?
Also add a comment to explain.

Reply:
Good point, looks better now